src/integration/resources/types.go

```diff
@@ -69,10 +70,16 @@ type Coordinator interface {
 	RunQuery(verifier ResponseVerifier, query string, headers Headers) error
 	// InstantQuery runs an instant query with provided headers
 	InstantQuery(req QueryRequest, headers Headers) (model.Vector, error)
+	// InstantQueryWithEngine runs an instant query with provided headers and the specified
+	// query engine.
+	InstantQueryWithEngine(req QueryRequest, engine options.QueryEngine, headers Headers) (model.Vector, error)
```

Comment:
interesting, can't the engine be specified as a header?

Reply:
Ah yeah, it can, but this method allows us to test the prom and m3query routes directly rather than using the the generic route and then sending to the appropriate handler based on the header. Some tests looked like they wanted to hit the route directly so exposed this endpoint